src/llm_vm/onsite_llm.py

```diff
@@ -303,6 +304,36 @@ def model_loader(self):
         return LlamaForCausalLM.from_pretrained(self.model_uri)
     def tokenizer_loader(self):
         return LlamaTokenizer.from_pretrained(self.model_uri)
+    
+@RegisterModelClass("quantized-llama")
+class Quantized_Llama(Base_Onsite_LLM):
+    """
+    Class for running quantized Llama instances that use GGML
+
+    Attributes:
+        model_uri (str): Hugging Face Endpoint for LLM
+        model (LLM): The large language model. CTransformers includes its own tokenizer in the model
+
+    Methods:
+        model_loader: Loads the LLM into memory
+        tokenizer_loader: Does nothing. CTransformers includes its own tokenizer so this is unnecessary
+        generate: Generates a response from a given prompt with the loaded LLM and tokenizer
+    """
+    model_uri="TheBloke/LLaMa-7B-GGML"
+
+    def model_loader(self):
+        # This file specifically is the smallest model.
+        return AutoModelForCausalLM.from_pretrained(self.model_uri, model_file="llama-7b.ggmlv3.q2_K.bin")
```

Comment:
question: why are you specifying both file and uri?

Reply:
or is it file in the repo?

Reply:
There are several versions of the model, with varying quantizations. The one I listed is the 2bit model, I just used the smallest model they have.

Reply:
i guess ggml repos tend to have more than one model to pick from
ok cool, sounds good

Reply:
we do need some way to overrride model file name though, i think you need to setup an __init__ that takes the right kwargs with default values so you can thread stuff into that space

Reply:
^ working on that now